lib/application/logic/saved_schedule_state.dart

```diff
@@ -0,0 +1,23 @@
+part of 'saved_schedule_cubit.dart';
+
+class SavedScheduleState extends Equatable {
+  final List<ScheduleItem> scheduleItems;
+  final List<ScheduleItem> savedItems;
+  final bool loading;
+  final dynamic error;
+  final DateTime currentDate;
+  final ScheduleItemCategory currentCategory;
+  final List<DateTime> allDates;
+
+  const SavedScheduleState(
+      {this.scheduleItems = const [],
+      this.savedItems = const [],
+      this.loading = true,
+      this.error,
+      required this.currentDate,
+      required this.allDates,
+      this.currentCategory = ScheduleItemCategory.all});
+
```

Comment:
Dodajte si ovdje poseban konstruktor tipa `SavedScheduleState.initial()` pa onda stavite da taj konstruktor uvijek returna ono što koristite u Cubitu kao počeno stanje i što u Cubitu unutar `readAllSavedItems` metode prvo emitate.